File: RenderSample/Common/Material.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SampleCommon
{
	public class Material
	{
		// 自发光颜色
		public Color mEmissive;
		// 环境光系数。
		public float mKA;
		// 漫反射颜色
		public Color mDiffuse;

		// TODO:增加镜面高光的支持。

		public Color Emissive
		{
			get { return mEmissive; }
			set { mEmissive = value; }
		}

		public float KA
		{
			get { return mKA; }
			set { mKA = KA; }
		}

		public Color Diffuse
		{
			get { return mDiffuse; }
			set { mDiffuse = value; }
		}

		public Material(Color emissive, float ka, Color diffuse)
		{
			mEmissive = emissive;
			mKA = ka;
			mDiffuse = diffuse;
		}
	}
}

File: WindowsGraphics/WindowsGraphics/Material.cpp
#include "Material.h"

Material::Material() :
	Ka(0.0), Kd(0.0), Ks(0.0)
{/*empty*/}

Material::Material(double _Ka, 
					 double _Kd,
					 double _Ks,
					 int _n)
	: Ka(_Ka), Kd(_Kd), Ks(_Ks), n(_n)
{/*empty*/}

Material::~Material()
{/*empty*/}